src/v4/hooks/dca/DCAHook.sol

```diff
@@ -0,0 +1,479 @@
+        // Approach: loop once to total output; for each allocation, loop outputs to sum matching recipient.
+        for (uint256 i = 0; i < outputs.length; i++) {
+            // token already checked equals intent.outputToken in _beforeTokenTransfer
+            totalOutput += outputs[i].amount;
+        }
+
+        for (uint256 i = 0; i < intent.outputAllocations.length; i++) {
+            address rcpt = intent.outputAllocations[i].recipient;
+            uint256 expected = Math.mulDiv(totalOutput, intent.outputAllocations[i].basisPoints, 10000);
+            uint256 actual = 0;
+            for (uint256 j = 0; j < outputs.length; j++) {
+                if (outputs[j].recipient == rcpt) actual += outputs[j].amount;
+            }
+            if (intent.isExactIn) {
+                // Allow ±1 wei for integer division rounding
+                if (!(actual + 1 >= expected && actual <= expected + 1)) {
+                    revert AllocationMismatch(rcpt, actual, expected);
+                }
+            } else {
+                if (actual != expected) {
+                    revert AllocationMismatch(rcpt, actual, expected);
+                }
+            }
+        }
+
+        if (intent.isExactIn) {
+            // total output produced must meet the limit
+            if (totalOutput < cosignerData.limitAmount) {
+                revert InsufficientOutput(totalOutput, cosignerData.limitAmount);
+            }
+        } else {
+            // exact output must be matched
+            if (totalOutput != cosignerData.execAmount) {
+                revert WrongTotalOutput(totalOutput, cosignerData.execAmount);
+            }
+        }
+    }
+
+    /// @notice Updates the execution state after successful validation
+    /// @dev Updates counters, totals, timestamps and nonce for the DCA intent
+    /// @param intentId The unique identifier for this DCA intent
+    /// @param inputAmount The amount of input tokens being executed
+    /// @param outputs The output tokens being distributed
+    /// @return totalInputExecuted The cumulative input amount after this execution
+    /// @return totalOutputExecuted The cumulative output amount after this execution
+    function _updateExecutionState(bytes32 intentId, uint256 inputAmount, OutputToken[] memory outputs)
+        internal
+        returns (uint256 totalInputExecuted, uint256 totalOutputExecuted)
+    {
+        // Use memory to reduce SSTOREs
+        DCAExecutionState memory state = executionStates[intentId];
+
+        // Calculate total output amount
+        uint256 totalOutput = 0;
+        for (uint256 i = 0; i < outputs.length; i++) {
+            totalOutput += outputs[i].amount;
+        }
+
+        // Update state in memory
+        state.executedChunks++;
+        state.lastExecutionTime = block.timestamp;
+        state.totalInputExecuted += inputAmount;
+        state.totalOutput += totalOutput;
+        state.nextNonce++;
```

Comment:
Since `nextNonce` and `executedChunks` are always the same, can we just use one or the other?